### ISSUE TITLE
Macro Precision computation is wrong when ignore_index is used

### ISSUE THREAD

**Issue by mpBarbato** [Lightning-AI/torchmetrics#1692]
## 🐛 Bug

When ignore_index is used in MulticlassPrecision, it zeros out the metric on the class to ignore but it still includes the value in the average computation, leading to wrong results.

### To Reproduce

```python
import torch
from torchmetrics.classification import MulticlassPrecision

# simulate the output of a perfect predictor (i.e. preds == target)
target = torch.tensor([0, 1, 2, 0, 1, 2])
preds  = target.clone()

metric = MulticlassPrecision(num_classes=3, average='none', ignore_index=0)
res = metric(preds, target)
print(res)
# it prints [0., 1., 1.]

metric = MulticlassPrecision(num_classes=3, average='macro', ignore_index=0)
res = metric(preds, target)
print(res)
# it prints 0.6667 instead of 1
```

### Expected behavior

It should not consider the ignored class in the average computation.

### Environment

- TorchMetrics version: 0.11.4 installed from pip
- Python & PyTorch Version: 3.7.9 - 1.11.0+cu113
- Any other relevant information such as OS (e.g., Linux): Linux Mint 18.3 Sylvia

**Comment by adamjstewart** [comment]
Duplicate of #1691
